framework/source/class/qx/ui/window/Window.js

```diff
@@ -929,6 +1055,40 @@ qx.Class.define("qx.ui.window.Window",
       }
     },
 
+    // overridden
+    _applyCenterWhen : function(value, old)
+    {
+      var             parent;
+
+      // Remove prior listener for centering on appear
+      if (this.__centeringAppearId !== null) {
+        this.removeListenerById(this.__centeringAppearId);
+        this.__centeringAppearId = null;
+      }
+
+      // Remove prior listener for centering on resize
+      if (this.__centeringResizeId !== null) {
+        parent.removeListenerById(this.__centeringResizeId);
+        this.__centeringResizeId = null;
+      }
+
+      // If we are to center on appear, arrange to do so
+      if (value.indexOf("appear") != -1) {
+        this.__centeringAppearId =
+          this.addListener("appear", this.center, this);
+      }
+
+      // If we are to center on resize, arrange to do so
+      if (value.indexOf("resize") != -1) {
+        parent = this.getLayoutParent();
```

Comment:
It could theoretically happen that the layout parent changes, I think. We'd then have to hook into the ```setLayoutParent``` method, remove the event listener from the old parent and attach the listener to the new one. This gets simpler if we always attach a resize listener to the parent. All this can be done in ```setLayoutParent```